packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -287,15 +284,59 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
         return DEFAULT_TERMINAL_RENDERER_TYPE;
     }
 
-    showHoverMessage(x: number, y: number, message: string): void {
-        this.hoverMessage.innerText = message;
-        this.hoverMessage.style.display = 'inline';
-        this.hoverMessage.style.top = `${y - 30}px`;
-        this.hoverMessage.style.left = `${x - 60}px`;
+    protected initializeLinkHover(): void {
+        this.linkHover = document.createElement('div');
+        this.linkHover.style.position = 'fixed';
+        this.linkHover.style.color = 'var(--theia-editorHoverWidget-foreground)';
+        this.linkHover.style.backgroundColor = 'var(--theia-editorHoverWidget-background)';
+        this.linkHover.style.borderColor = 'var(--theia-editorHoverWidget-border)';
+        this.linkHover.style.borderWidth = '0.5px';
+        this.linkHover.style.borderStyle = 'solid';
+        this.linkHover.style.padding = '5px';
+        // Above the xterm.js canvas layers:
+        // https://github.com/xtermjs/xterm.js/blob/ff790236c1b205469f17a21246141f512d844295/src/renderer/Renderer.ts#L41-L46
+        this.linkHover.style.zIndex = '10';
+        // Initially invisible:
+        this.linkHover.style.display = 'none';
+
+        this.linkHoverButton = document.createElement('a');
+        this.linkHoverButton.textContent = this.linkHoverMessage();
+        this.linkHoverButton.style.cursor = 'pointer';
+        this.linkHover.appendChild(this.linkHoverButton);
+
+        const cmdCtrl = isOSX ? 'cmd' : 'ctrl';
+        const cmdHint = document.createTextNode(` (${nls.localizeByDefault(`${cmdCtrl} + click`)})`);
+        this.linkHover.appendChild(cmdHint);
+
+        const onMouseEnter = (mouseEvent: MouseEvent) => this.onMouseEnterLinkHoverEmitter.fire(mouseEvent);
+        this.linkHover.addEventListener('mouseenter', onMouseEnter);
+        this.toDispose.push(Disposable.create(() => this.linkHover.removeEventListener('mouseenter', onMouseEnter)));
+
+        const onMouseLeave = (mouseEvent: MouseEvent) => this.onMouseLeaveLinkHoverEmitter.fire(mouseEvent);
+        this.linkHover.addEventListener('mouseleave', onMouseLeave);
+        this.toDispose.push(Disposable.create(() => this.linkHover.removeEventListener('mouseleave', onMouseLeave)));
+
+        this.node.appendChild(this.linkHover);
+    }
+
+    showLinkHover(invokeAction: (event: MouseEvent) => void, x: number, y: number, message?: string): void {
+        const mouseY = this.lastMousePosition?.y ?? y;
+        const mouseX = this.lastMousePosition?.x ?? x;
```

Comment:
Suggestion: While the current behavior already works quite well, I noticed that vscode always renders the hover right above the selected link. This makes it less dependent on the actual cursor position, which can lead to minor issues such as these (there's a link hidden unter the hover):
![image](https://user-images.githubusercontent.com/4377073/184151655-c762553b-0b05-4e2f-b6dd-1ae38287eea4.png)

Do you think we could implement something similar?

_vscode_

![image](https://user-images.githubusercontent.com/4377073/184151956-cb58de57-c683-43bd-9719-d3f12f9034e4.png)

Reply:
Great suggestion! VS Code indeed does a nicer job here in positioning the hover compared to my PR. I've actually looked into the implementation of VS Code already as I wanted to place the hover similarly to how it is done in VS Code. Eventually, however I didn't do it for now, as it would require quite some extra code (if I didn't miss anything):

* The computation of the hover position is in [`CellHoverTarget`](https://github.com/microsoft/vscode/blob/a71a077d81f970e8fe3ddad753ac045a7dbaa9eb/src/vs/workbench/contrib/terminal/browser/widgets/terminalHoverWidget.ts#L66)
* Additionally, we need to obtain the actual cell dimensions, as done in the [`TerminalLinkManager`](https://github.com/microsoft/vscode/blob/a71a077d81f970e8fe3ddad753ac045a7dbaa9eb/src/vs/workbench/contrib/terminal/browser/links/terminalLinkManager.ts#L213)
* And we need the add the [`IXtermCore`](https://github.com/microsoft/vscode/blob/a71a077d81f970e8fe3ddad753ac045a7dbaa9eb/src/vs/workbench/contrib/terminal/browser/xterm-private.d.ts#L1) interface

On a first glance this didn't seem worth it for this change. However, if you prefer, I can give it a second look and try it out.

Reply:
It's fine, it's nothing that stops this from being merged. I just hoped it'd be something that could be easily added, but the changes outlined are out of scope for this PR.